File: src/SOP8/Base/History/Base.History.IBLL/Request/RequestExcelDetectHistory.cs
using System.Collections.Generic;

namespace Base.History.IBLL.Request
{
    using Models.History;

    // 전체 다운로드
    public class RequestExcelDetectHistory
    {
        private int m_nBeginYear = -1;
        private int m_nBeginMonth = -1;
        private int m_nBeginDay = -1;
        private int m_nEndYear = -1;
        private int m_nEndMonth = -1;
        private int m_nEndDay = -1;
        private int? m_sensorType = null;
        private List<int> m_sensorSubTypes = null;
        private int? m_buildingGroupNo = null;
        private int? m_buildingNo = null;
        private int? m_zoneNo = null;
        private int? m_sensorNo = null;
        private int? m_siteNo = null;

        private List<SensorTypeData> m_sensorTypeDatas = null;

        private bool m_useSensorTypeName = true;
        private bool m_useSensorName = true;
        private bool m_useLocationName = true;
        private bool m_useDetectStatus = false;
        private bool m_useClearType = true;
        private bool m_useAlarmDepthName = true;
        private bool m_useSopName = false;
        private bool m_useMemo = false;
        private string m_strSubjectName = null;

        public int BeginYear
        {
            get { return m_nBeginYear; }
            set { m_nBeginYear = value; }
        }

        public int BeginMonth
        {
            get { return m_nBeginMonth; }
            set { m_nBeginMonth = value; }
        }

        public int BeginDay
        {
            get { return m_nBeginDay; }
            set { m_nBeginDay = value; }
        }

        public int EndYear
        {
            get { return m_nEndYear; }
            set { m_nEndYear = value; }
        }

        public int EndMonth
        {
            get { return m_nEndMonth; }
            set { m_nEndMonth = value; }
        }

        public int EndDay
        {
            get { return m_nEndDay; }
            set { m_nEndDay = value; }
        }

        public int? BuildingGroupNo
        {
            get { return m_buildingGroupNo; }
            set { m_buildingGroupNo = value; }
        }

        public int? BuildingNo
        {
            get { return m_buildingNo; }
            set { m_buildingNo = value; }
        }

        public int? ZoneNo
        {
            get { return m_zoneNo; }
            set { m_zoneNo = value; }
        }

        public int? SensorNo
        {
            get { return m_sensorNo; }
            set { m_sensorNo = value; }
        }

        public int? SensorType
        {
            get { return m_sensorType; }
            set { m_sensorType = value; }
        }

        public List<int> SensorSubTypes
        {
            get { return m_sensorSubTypes; }
            set { m_sensorSubTypes = value; }
        }

        public int? SiteNo
        {
            get { return m_siteNo; }
            set { m_siteNo = value; }
        }

        public List<SensorTypeData> SensorTypeDatas
        {
            get { return m_sensorTypeDatas; }
            set { m_sensorTypeDatas = value; }
        }

        public bool UseSensorTypeName
        {
            get { return m_useSensorTypeName; }
            set { m_useSensorTypeName = value; }
        }

        public bool UseSensorName
        {
            get { return m_useSensorName; }
            set { m_useSensorName = value; }
        }

        public bool UseLocationName
        {
            get { return m_useLocationName; }
            set { m_useLocationName = value; }
        }

        public bool UseDetectStatus
        {
            get { return m_useDetectStatus; }
            set { m_useDetectStatus = value; }
        }

        public bool UseClearType
        {
            get { return m_useClearType; }
            set { m_useClearType = value; }
        }

        public bool UseAlarmDepthName
        {
            get { return m_useAlarmDepthName; }
            set { m_useAlarmDepthName = value; }
        }

        public bool UseSopName
        {
            get { return m_useSopName; }
            set { m_useSopName = value; }
        }

        public bool UseMemo
        {
            get { return m_useMemo; }
            set { m_useMemo = value; }
        }

        public string SubjectName
        {
            get { return m_strSubjectName; }
            set { m_strSubjectName = value; }
        }

        public virtual RequestSensorDetectHistory ToRequestSensorDetectHistory()
        {
            RequestSensorDetectHistory request = MakeRequestSensorDetectHistory();

            request.BeginYear = this.BeginYear;
            request.BeginMonth = this.BeginMonth;
            request.BeginDay = this.BeginDay;
            request.EndYear = this.EndYear;
            request.EndMonth = this.EndMonth;
            request.EndDay = this.EndDay;
            request.BuildingGroupNo = this.BuildingGroupNo;
            request.BuildingNo = this.BuildingNo;
            request.ZoneNo = this.ZoneNo;
            request.SensorNo = this.SensorNo;
            request.SensorType = this.SensorType;
            request.SensorSubTypes = this.SensorSubTypes;
            request.SiteNo = this.SiteNo;
            request.SensorTypeDatas = this.SensorTypeDatas;
            request.UseSensorTypeName = this.UseSensorTypeName;
            request.UseSensorName = this.UseSensorName;
            request.UseLocationName = this.UseLocationName;
            request.UseDetectStatus = this.UseDetectStatus;
            request.UseClearType = this.UseClearType;
            request.UseAlarmDepthName = this.UseAlarmDepthName;
            request.UseSopName = this.UseSopName;
            request.UseMemo = this.UseMemo;

            return request;
        }

        protected virtual RequestSensorDetectHistory MakeRequestSensorDetectHistory()
        {
            return new RequestSensorDetectHistory();
        }
    }
}

File: src/SOP8/Egovframework/Base/base-history-ibll/src/main/java/egovframework/base/history/ibll/request/RequestExcelDetectHistory.java
package egovframework.base.history.ibll.request;

import egovframework.base.history.ibll.models.history.SensorTypeData;

import java.util.List;

public class RequestExcelDetectHistory {

    private int beginYear = -1;
    private int beginMonth = -1;
    private int beginDay = -1;
    private int endYear = -1;
    private int endMonth = -1;
    private int endDay = -1;

    private Integer sensorType = null;
    private List<Integer> sensorSubTypes = null;
    private Integer buildingGroupNo = null;
    private Integer buildingNo = null;
    private Integer zoneNo = null;
    private Integer sensorNo = null;
    private Integer siteNo = null;

    private List<SensorTypeData> sensorTypeDatas = null;

    private boolean useSensorTypeName = true;
    private boolean useSensorName = true;
    private boolean useLocationName = true;
    private boolean useDetectStatus = false;
    private boolean useClearType = true;
    private boolean useAlarmDepthName = true;
    private boolean useSopName = false;
    private boolean useMemo = false;

    // Getters and Setters

    public int getBeginYear() {
        return beginYear;
    }

    public void setBeginYear(int beginYear) {
        this.beginYear = beginYear;
    }

    public int getBeginMonth() {
        return beginMonth;
    }

    public void setBeginMonth(int beginMonth) {
        this.beginMonth = beginMonth;
    }

    public int getBeginDay() {
        return beginDay;
    }

    public void setBeginDay(int beginDay) {
        this.beginDay = beginDay;
    }

    public int getEndYear() {
        return endYear;
    }

    public void setEndYear(int endYear) {
        this.endYear = endYear;
    }

    public int getEndMonth() {
        return endMonth;
    }

    public void setEndMonth(int endMonth) {
        this.endMonth = endMonth;
    }

    public int getEndDay() {
        return endDay;
    }

    public void setEndDay(int endDay) {
        this.endDay = endDay;
    }

    public Integer getSensorType() {
        return sensorType;
    }

    public void setSensorType(Integer sensorType) {
        this.sensorType = sensorType;
    }

    public List<Integer> getSensorSubTypes() {
        return sensorSubTypes;
    }

    public void setSensorSubTypes(List<Integer> sensorSubTypes) {
        this.sensorSubTypes = sensorSubTypes;
    }

    public Integer getBuildingGroupNo() {
        return buildingGroupNo;
    }

    public void setBuildingGroupNo(Integer buildingGroupNo) {
        this.buildingGroupNo = buildingGroupNo;
    }

    public Integer getBuildingNo() {
        return buildingNo;
    }

    public void setBuildingNo(Integer buildingNo) {
        this.buildingNo = buildingNo;
    }

    public Integer getZoneNo() {
        return zoneNo;
    }

    public void setZoneNo(Integer zoneNo) {
        this.zoneNo = zoneNo;
    }

    public Integer getSensorNo() {
        return sensorNo;
    }

    public void setSensorNo(Integer sensorNo) {
        this.sensorNo = sensorNo;
    }

    public Integer getSiteNo() {
        return siteNo;
    }

    public void setSiteNo(Integer siteNo) {
        this.siteNo = siteNo;
    }

    public List<SensorTypeData> getSensorTypeDatas() {
        return sensorTypeDatas;
    }

    public void setSensorTypeDatas(List<SensorTypeData> sensorTypeDatas) {
        this.sensorTypeDatas = sensorTypeDatas;
    }

    public boolean isUseSensorTypeName() {
        return useSensorTypeName;
    }

    public void setUseSensorTypeName(boolean useSensorTypeName) {
        this.useSensorTypeName = useSensorTypeName;
    }

    public boolean isUseSensorName() {
        return useSensorName;
    }

    public void setUseSensorName(boolean useSensorName) {
        this.useSensorName = useSensorName;
    }

    public boolean isUseLocationName() {
        return useLocationName;
    }

    public void setUseLocationName(boolean useLocationName) {
        this.useLocationName = useLocationName;
    }

    public boolean isUseDetectStatus() {
        return useDetectStatus;
    }

    public void setUseDetectStatus(boolean useDetectStatus) {
        this.useDetectStatus = useDetectStatus;
    }

    public boolean isUseClearType() {
        return useClearType;
    }

    public void setUseClearType(boolean useClearType) {
        this.useClearType = useClearType;
    }

    public boolean isUseAlarmDepthName() {
        return useAlarmDepthName;
    }

    public void setUseAlarmDepthName(boolean useAlarmDepthName) {
        this.useAlarmDepthName = useAlarmDepthName;
    }

    public boolean isUseSopName() {
        return useSopName;
    }

    public void setUseSopName(boolean useSopName) {
        this.useSopName = useSopName;
    }

    public boolean isUseMemo() {
        return useMemo;
    }

    public void setUseMemo(boolean useMemo) {
        this.useMemo = useMemo;
    }
}